### ISSUE TITLE
Drush could not find the docroot. 

### ISSUE THREAD

**Issue by t14** [acquia/blt#2399]
My system information:
* Operating system type: ubuntu/trusty64
* BLT version: ^8.3

Output of `blt doctor`:
```
 blt doctor
[warning] The xDebug extension is loaded. This will significantly decrease performance.
Attempting to run doctor on host machine...
Drush could not find the docroot.

```

When I run this command:
```
blt setup
```

I get the following output:
```
[warning] The xDebug extension is loaded. This will significantly decrease performance.
Setting up local environment for site '' using drush alias @
> setup:build
>> setup:behat
>> setup:composer:install
Cannot create cache directory /home/vagrant/.composer/cache/repo/https---packagist.org/, or directory is not writable. Proceeding without cache
Cannot create cache directory /home/vagrant/.composer/cache/files/, or directory is not writable. Proceeding without cache
Cannot create cache directory /home/vagrant/.composer/cache/repo/https---packages.drupal.org-8/, or directory is not writable. Proceeding without cache
Cannot create cache directory /home/vagrant/.composer/cache/repo/https---asset-packagist.org/, or directory is not writable. Proceeding without cache
Gathering patches for root package.
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
>> setup:git-hooks
Installing pre-commit git hook...
Installing commit-msg git hook...
>> setup:settings
>> frontend
> frontend:setup
> frontend:build
> setup:hash-salt
Hash salt already exists.
> setup:drupal:install
[warning] /var/www/topgearD8/docroot/sites//settings/local.settings.php is missing.
 Do you want to generate this required settings file(s)? (y/n) y
[warning] The xDebug extension is loaded. This will significantly decrease performance.
[error]  MySql is not available. Please run `blt doctor` to diagnose the issue. 
[error]  Command `setup:drupal:install ` exited with code 1. 

```

And I expected this to happen:
Drupal to be installed

For various reason I am unable to use the Drupal VM, I have my own LAMP stack already setup

**Comment by grasmash** [comment]
Can you share your directory structure and the output of `drush status` ?

**Comment by t14** [comment]
Drush status?
 
```
 PHP executable         :  /usr/bin/php                                    
 PHP configuration      :  /etc/php/5.6/cli/php.ini                        
 PHP OS                 :  Linux                                           
 Drush script           :  /var/www/D8/vendor/drush/drush/drush.php 
 Drush version          :  8.1.15                                          
 Drush temp directory   :  /tmp                                            
 Drush configuration    :                                                  
 Drush alias files      :  
```

```
# ls command from directory /var/www/D8
blt  composer.json  composer.lock  config  docroot  drush  patches  phpcs.xml.dist  README.md  salt.txt  tests	vendor
```

**Comment by grasmash** [comment]
BLT is correctly reporting a real issue--drush cannot find the drupal root. If you resolve the Drush  issue, then BLT will work.

My best guess is that some files or folders are missing. Perhaps files in `docroot/core`? Can you re-install Composer dependencies?

**Comment by t14** [comment]
when I run `blt doctor` from `docroot` it says
`You must run this command from within a BLT-generated project repository.`

Am I suppose to have all Drupal source code in docroot after running `composer create-project --no-interaction acquia/blt-project MY_PROJECT` ?

My docroot currently looks like this with no actual drupal source code after running the `composer create-project `

```
$ ls
autoload.php  modules  profiles  sites	themes
```

**Comment by grasmash** [comment]
Yes, you should. It looks like you encountered an error when executing the composer command.

**Comment by grasmash** [comment]
Try ‘composer create-project acquia/blt-project:8.x-dev --no-interaction my-project’

**Comment by jayakrishnanj** [comment]
`[error]  MySql is not available.`
@t14 , Could you please verify Mysql is running and accessible?

**Comment by grasmash** [comment]
> Cannot create cache directory /home/vagrant/.composer/cache/repo/https---packagist.org/, or directory is not writable. Proceeding without cache

Also, please run all commands __outside__ the VM.

**Comment by t14** [comment]
@jayakrishnanj yes mysql is working
@grasmash  I ran this outside my VM and it worked fine.
Looks like the issue was with composer
Composer was throwing this error  
`  [ErrorException]                                   
  proc_open(): fork failed - Cannot allocate memory`

So I followed the instructions in the link below and created a swap file on my VM and it fixed the issue everything else ran fine after that. In my case I had to create a 4GB swap file before it worked. Seems like composer is very resource intense
` https://getcomposer.org/doc/articles/troubleshooting.md#proc-open-fork-failed-errors`

Thank you both for taking the time to help me out
